docroot/modules/humsci/hs_paragraph_types/hs_paragraph_types.module

```diff
@@ -31,3 +33,28 @@ function hs_paragraph_types_preprocess_field__field_hs_collection_items(&$variab
   $variables['attributes']['class'][] = "item-per-row--$items_per_row";
   $variables['#attached']['library'][] = 'hs_paragraph_types/hs_collection';
 }
+
+/**
+ * Implements hook_field_widget_complete_WIDGET_TYPE_form_alter().
+ */
+function hs_paragraph_types_field_widget_complete_paragraphs_form_alter(array &$field_widget_form, FormStateInterface $form_state, array $context) {
+  if (
+    isset($field_widget_form['widget']['add_more']['add_more_button_hs_hero_image'], $field_widget_form['widget']['#field_name']) &&
+    $field_widget_form['widget']['#field_name'] == 'field_hs_carousel_slides'
+  ) {
+    $new_label = 'Banner image with text box';
+    // Change the add button label for field_hs_carousel_slides.
+    $field_widget_form['widget']['add_more']['add_more_button_hs_hero_image']['#value'] = t('Add @type', [
+      '@type' => $new_label,
+    ]);
+    // Change the paragraph type label for hs_hero_image embedded in hs_carousel.
+    foreach ($field_widget_form['widget'] as $key => $element) {
+      // Only alter markup in field widget elements.
+      if (!is_int($key)) {
+        continue;
+      }
+      $element['top']['type']['label']['#markup'] = $new_label;
+      $field_widget_form['widget'][$key] = $element;
```

Comment:
Don't you need to loop by reference if you want to change the label? `&$element`

Reply:
Oh now that i read it more closely, it would be alot better to loop by reference and remove line 57 entirely.
Also, a better way (and more consistent with core code) to loop on the keys is to use `\Drupal\Core\Render\Element::children($field_widget_form['widget'])`. This will give you the keys and you can check for values in the arrays.

Reply:
@pookmish Fixed. PR is ready for review again.